Source/NSubstitute/Core/GetCallSpec.cs

```diff
@@ -1,26 +1,45 @@
-﻿namespace NSubstitute.Core
+﻿using System;
+
+namespace NSubstitute.Core
 {
     public class GetCallSpec : IGetCallSpec
     {
-        private readonly ICallStack _callStack;
+        private readonly ICallCollection _callCollection;
         private readonly IPendingSpecification _pendingSpecification;
         private readonly ICallSpecificationFactory _callSpecificationFactory;
         private readonly ICallActions _callActions;
 
-        public GetCallSpec(ICallStack callStack, IPendingSpecification pendingSpecification,
+        public GetCallSpec(ICallCollection callCollection, IPendingSpecification pendingSpecification,
             ICallSpecificationFactory callSpecificationFactory, ICallActions callActions)
         {
-            _callStack = callStack;
+            _callCollection = callCollection;
             _pendingSpecification = pendingSpecification;
             _callSpecificationFactory = callSpecificationFactory;
             _callActions = callActions;
         }
 
-        public ICallSpecification FromLastCall(MatchArgs matchArgs)
+        public ICallSpecification FromPendingSpecification(MatchArgs matchArgs)
         {
-            return _pendingSpecification.HasPendingCallSpec()
-                               ? FromExistingSpec(_pendingSpecification.UseCallSpec(), matchArgs)
-                               : FromCall(_callStack.Pop(), matchArgs);
+            if (!_pendingSpecification.HasPendingCallSpecInfo())
+            {
+                throw new InvalidOperationException("No pending specification or previous call info.");
```

Comment:
There are a few `throw new InvalidOperationException(...)` calls like this. If these are things that can happen in the normal course of use I think it would be better to use/reuse a custom SubstituteException with an explanation of what has gone wrong, and how the person using the library can modify their code to fix it.